public/components/ClearCacheModal/ClearCacheModal.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { CoreStart } from "opensearch-dashboards/public";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+import { ServicesContext } from "../../services";
+import { CoreServicesContext } from "../core_services";
+import { IndexOpBlocksType } from "../../utils/constants";
+import {
+  getErrorMessage,
+  aliasBlockedPredicate,
+  indexBlockedPredicate,
+  dataStreamBlockedPredicate,
+  filterBlockedItems,
+} from "../../utils/helpers";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+
+interface ClearCacheModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  onClose: () => void;
+  type: string;
+}
+
+export default function ClearCacheModal<T>(props: ClearCacheModalProps<T>) {
+  const { onClose, visible, selectedItems, type } = props;
+  const [hint, setHint] = useState("");
+  const [blockHint, setBlockHint] = useState("");
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+
+  useEffect(() => {
+    const indexBlocksTypes = [
+      IndexOpBlocksType.Closed,
+      IndexOpBlocksType.MetaData,
+      IndexOpBlocksType.ReadOnly,
+      IndexOpBlocksType.ReadOnlyAllowDelete,
+    ];
+    if (!!services && !!visible) {
+      switch (type) {
+        case "data streams":
```

Comment:
It would be better that we use an enum for these strings.

Reply:
Sure, I'll change it.